packages/task/src/browser/task-service.ts

```diff
@@ -70,7 +79,8 @@ export class TaskService implements TaskConfigurationClient {
     protected cachedRecentTasks: TaskConfiguration[] = [];
     protected runningTasks = new Map<number, {
         exitCode: Deferred<number | undefined>,
-        terminateSignal: Deferred<string | undefined>
+        terminateSignal: Deferred<string | undefined>,
+        isBackgroundTaskEnded: Deferred<boolean | undefined>
```

Comment:
dumb question: why is `isBackgroundTaskEnded` a `Deferred` ? what prevents us simply using a boolean ?

Reply:
It must be a promise ( like the 'exitCode') because after start running the task, we wait for the task to exit (or to be activated in the backgroud) so we wait till this promise resolved